packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -1400,6 +1400,145 @@ export abstract class TargetObjectKeyFormat {
+   */
+  readonly kmsKey?: kms.IKey;
+
+  /**
+   * The storage class to use when replicating objects, such as S3 Standard or reduced redundancy.
+   *
+   * @default - The storage class of the source object
+   */
+  readonly storageClass?: StorageClass;
+
+  /**
+   * Specifies whether Amazon S3 replicates objects created with server-side encryption using an AWS KMS key stored in AWS Key Management Service.
+   *
+   * @default false
+   */
+  readonly sseKmsEncryptedObjects?: boolean;
+
+  /**
+   * Specifies whether Amazon S3 replicates modifications on replicas.
+   *
+   * @default false
+   */
+  readonly replicaModifications?: boolean;
+
+  /**
+   * The priority indicates which rule has precedence whenever two or more replication rules conflict.
+   *
+   * Amazon S3 will attempt to replicate objects according to all replication rules.
+   * However, if there are two or more rules with the same destination bucket,
+   * then objects will be replicated according to the rule with the highest priority.
+   *
+   * The higher the number, the higher the priority.
+   *
+   * It is essential to specify priority explicitly when the replication configuration has multiple rules.
+   *
+   * @default 0
+   */
+  readonly priority?: number;
+
+  /**
+   * Specifies whether Amazon S3 replicates delete markers.
+   *
+   * @default false
+   */
+  readonly deleteMarkerReplication?: boolean;
+
+  /**
+   * A unique identifier for the rule.
+   *
+   * The maximum value is 255 characters.
+   *
+   * @default - auto generated random ID
+   */
+  readonly id?: string;
+
+  /**
+   * An object key name prefix that identifies the object or objects to which the rule applies.
+   *
+   * @default - applies to all objects
+   */
+  readonly prefixFilter?: string;
+
+  /**
+   * The tag array used for tag filters.
+   *
+   * The rule applies only to objects that have the tag in this set.
+   *
+   * @default - applies to all objects
+   */
+  readonly tagFilter?: Tag[];
+}
+
+/**
+ * A container for information about the replication destination.
+ */
+export class ReplicationDestination {
+  /**
+   * Replicate to another bucket in the same account.
+   *
+   * @param bucket the destination bucket
+   */
+  public static sameAccount(bucket: IBucket): ReplicationDestination {
+    return new ReplicationDestination(bucket);
+  }
+
+  /**
+   * Replicate to another bucket in a different account.
+   *
+   * @param bucket the destination bucket
+   * @param account the account to replicate to
+   * @param accessControlTransition whether to want to change replica ownership to the AWS account that owns the destination bucket. The replicas are owned by same AWS account that owns the source object by default.
+   */
+  public static crossAccount(bucket: IBucket, account: string, accessControlTransition?: boolean): ReplicationDestination {
+    return new ReplicationDestination(bucket, account, accessControlTransition);
```

Comment:
I think there is a requirement that the destination bucket should add a statement in its bucket policy to give the permission to the replication role (example [here](https://repost.aws/knowledge-center/s3-cross-encrypted-replication)). As I don't see the specific logic in this PR I am wondering if this method should be there, throw an exception, or add the logic in the `renderReplicationConfiguration(...)` method.

Reply:
As you mentioned, when dealing with cross-account access, it's necessary to grant a bucket policy on the destination bucket. I think it’s great that this can be done through CDK.

However, I have the following concerns. Would it be possible to implement the proposed solution?

- I believe CDK does not have the authority to grant bucket policies on buckets in a different account.
- Similarly, I think CDK cannot retrieve the bucket policy of a bucket in a different account.

Reply:
>I believe CDK does not have the authority to grant bucket policies on buckets in a different account.

I think it is not a CDK constraint but an AWS profile permission constraint. 

> Similarly, I think CDK cannot retrieve the bucket policy of a bucket in a different account.

Correct. That's why I am proposing to throw an exception with a message saying that cross account replication setup is not supported by this construct. Which can let the community open a feature request. But if you keep this method as it is, you can already open a bug 😅

Reply:
Thank you very much for your clarification. 
Could you please confirm your intent with one of the following options?

- Do not implement the `crossAccount()` method in this PR and only support replication within the same account.
- Implement the `crossAccount()` method but display a warning related to the destination bucket policy using `Annotations` within the method.
- Other.

There is a possibility that I might not be interpreting your message correctly, so I would appreciate it if you could provide more specific instructions on how to proceed.

Reply:
Not sure this is what I initially meant but I think the best trade off would be to clearly document that method and explain that the user will have to configure the destination bucket policy to allow the replication role from the source account to replicate object in this destination bucket. And also add a warning message, which should ideally contain the full ARN of the replication role (this should simplify the customer experience).

---

If you want to put extra effort and  implement the destination bucket policy update if the type is `Bucket` it would be great as well.

Reply:
I've added some descriptions about destination bucket's policy.

```ts
/**
   * Replicate to another bucket in a different account.
   *
   * When performing cross-account replication,
   * you need to configure access permissions for the replication role in the bucket policy of the destination bucket.
   *
   * @see https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-walkthrough-2.html
```

And I've added `Annotations` about that.

```ts
        if (rule.destination.account !== undefined && rule.destination.account !== Stack.of(this).account) {
          Annotations.of(this).addInfo(`Cross-account S3 replication is set up. In the destination bucket's bucket policy, please grant access permissions from ${this.stack.resolve(role.roleArn)}.`);
        }
```

Reply:
I think I'd also prefer the bucket policy be added automatically. It is the CDK thing to do.

We can add the annotation if `addToBucketPolicy(): boolean` returns `false`, indicating that we have a referenced bucket.

Reply:
I've updated to add the bucket policy for the destination bucket.
I referenced [this document](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-walkthrough-2.html).

In addition, when enabling `accessControlTransition`, I also add this bucket policy.
https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-change-owner.html